GH-144: Switched to the hierarchical document symbols in the outline.

kittaakos opened the pull request:
~**WIP**: because it depends on https://github.com/eclipse/xtext-core/issues/882.~

Closes: #144.

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

JanKoehnlein commented on yang-lsp/gradle/versions.gradle:
```diff
@@ -1,8 +1,8 @@
 version = '0.1.0-SNAPSHOT'
 
 ext.versions = [
-    xtext: '2.15.0-SNAPSHOT',
```
Do we need the SNAPSHOT build?

kittaakos replied:
We need `2.16.0` for sure; it contains the changes for the hierarchical document symbols. I do not know if we can have `2.16.0` without the snapshot.

kittaakos commented:
> Have you tried with YANG Eclipse?

I have updated this PR; removed the workaround for https://github.com/eclipse/xtext-core/issues/882. How can I verify this with YANG Eclipse?
